spec.md

```diff
@@ -3614,11 +3681,14 @@ info = "@" , "[" , lineinfo, { ",", lineinfo }, "]" ;
 width = "<" , int_any , ">" ;
 type_ground = "Clock" | "Reset" | "AsyncReset"
             | ( "UInt" | "SInt" | "Analog" ) , [ width ] ;
+type_enum = "{|" , { field_enum } , "|}" ;
+field_enum = id, [ ":" , type ] ;
```

Comment:
1) Can enum data be const (probably?)?
2) Regardless, consider changing this to not include ref types-- either by splitting the `type` entry (e.g., `type_hw` or something) or just specifying `( type_ground | type_enum | type_aggregate )` here.  This aligns with how it's parsed (FIRRTLBaseType, presently, I believe).  Not critical to reject grammar-level all invalid types, but nice when possible.

Reply:
1) I don't see how if the enum itself isn't.  The "const on the bundle" implementation of const talked about elsewhere anticipates this as if the outer enum is constant, the inner data is, but not vice versa.  (e.g. const(non) makes sense, non(const) is really weird).
2) yea, no ref.  I mean, if you had a constant enum, then ref would work out I think, but let's not.